controllers/inference_services.go

```diff
@@ -295,6 +295,25 @@ func (r *TrustyAIServiceReconciler) patchKServe(ctx context.Context, instance *t
 			log.FromContext(ctx).Error(err, "InferenceService has service mesh annotation but DestinationRule CRD not found")
 		}
 
+		// Check if VirtualService CRD is present. If there's an error, don't proceed and return the error
```

Comment:
We should double check if the creation of the VirtualService is not producing any side effect like exposing the endpoint outside the namespace or similar.
I think we can do that with a follow-up

Reply:
Thanks @danielezonca, from my POV this isn't the case, as we discussed elsewhere.
Still I'm noting this for a more comprehensive review.